cunumeric/module.py

```diff
@@ -4671,6 +4671,19 @@ def cumprod(
     --------
     numpy.cumprod
 
+    Notes
+    -----
+    Due to the parallel nature of cuNumeric implementation, with floating point
+    and complex types, the accuracy of the results can differ compared to
+    NumPy. Furthermore when boundary values such as inf occur they might not
+    propagate as expected. Consider the hypothetical float32 array
+    ``[3e+37, 1, 100, 0.01]`` doing a cumprod on it should result in
+    ``[3e+37, 3e+37, inf, inf]`` in Numpy, however in cunumeric, if this array
+    is internally partitioned such that partition 0 has ``[3e+37, 1]``  and
+    partition 1 has ``[100, 0.01]``, the end result of the cumprod turns out
+    being ``[3e+37, 3e+37, inf, 3e+37]``, a result that albeit more acurate, is
+    often not expected.
```

Comment:
I think this message could be tightened up with plainer/simpler phrasing. Maybe something like:

>     CuNumeric's parallel implementation may yield different results from NumPy
>     with floating point and complex types. For example, when boundary values
>     such as inf occur they may not propagate as expected. Consider the float32
>     array ``[3e+37, 1, 100, 0.01]``. NumPy's cumprod will return a result of
>     ``[3e+37, 3e+37, inf, inf]``. However, cuNumeric might internally partition
>     the array such that partition 0 has ``[3e+37, 1]``  and partition 1 has 
>     ``[100, 0.01]``, returning the result ``[3e+37, 3e+37, inf, 3e+37]``.

Reply:
a3a8593fed3c1f0534e2d2dc251d75967b9790c3